Octokit.Tests.Integration/Clients/RepositoryCollaboratorClientTests.cs

```diff
@@ -136,4 +136,26 @@ public class TheIsCollaboratorMethod
             }
         }
     }
+
+    public class TheInviteMethod
+    {
+        [IntegrationTest]
+        public async Task CanInviteNewCollaborator()
+        {
+            var github = Helper.GetAuthenticatedClient();
+            var repoName = Helper.MakeNameWithTimestamp("public-repo");
+
+            using (var context = await github.CreateRepositoryContext(new NewRepository(repoName)))
+            {
+                var fixture = github.Repository.Collaborator;
+                var permission = new CollaboratorRequest();
+
+                // invite a collaborator
+                var response = await fixture.Invite(context.RepositoryOwner, context.RepositoryName, "maddin2016", permission);
```

Comment:
Ah right!  

The deserializer creates a "constructor cache" of all the constructors for the request/response objects, which it uses to instantiate the class, then it uses the protected setter properties to populate any fields it finds in the json payload, on the object it created.

But it can't use constructors that take parameters so all model classes actually require a parameterless constructor, which we have missed on the `RepositoryInvitation` class!